GameServerLib/GameObjects/Spells/Spell.cs

```diff
@@ -82,7 +82,7 @@ public virtual bool Cast(float x, float y, float x2, float y2, AttackableUnit u
 
             var stats = Owner.Stats;
             if (SpellData.ManaCost[Level] * (1 - stats.SpellCostReduction) >= stats.CurrentMana ||
-                State != SpellState.STATE_READY)
+                State != SpellState.STATE_READY || Owner.IsCastingSpell && Slot != 4 && Slot != 5)
```

Comment:
How do you read this condition?
```State != (SpellState.STATE_READY || Owner.IsCastingSpell) && (Slot != 4 && Slot != 5)```
or
```State != SpellState.STATE_READY || (Owner.IsCastingSpell && Slot != 4 && Slot != 5)```?
This expression should have clarifying parenthesis.

Reply:
It should be `... || (Owner.IsCastingSpell && Slot != 4 && Slot != 5)`